### PR TITLE
linkphase3: new package

### PR DESCRIPTION
**Pull request by akthoma**
adding new package linkphase3

### CODE REVIEW

File: var/spack/repos/builtin/packages/linkphase3/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    """Haplotype reconstruction in pedigreed populations."""
+
+    homepage = "https://github.com/tdruet/LINKPHASE3"
+    url      = "https://github.com/tdruet/LINKPHASE3"
+
+    version('2017-06-14', commit='559913593fc818bb1adb29796a548cf5bf323827',
+            git='https://github.com/tdruet/LINKPHASE3.git')
+
+    def install(self, spec, prefix):
+        gfortran = which('gfortran')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/linkphase3/package.py]: I guess, you should use `self.compiler.fc` here.

Comment (akthoma) [reply var/spack/repos/builtin/packages/linkphase3/package.py]: That doesn't work.
```
==> Executing phase: 'install'
==> Error: TypeError: 'syaml_str' object is not callable
TypeError: TypeError: 'syaml_str' object is not callable

/home/las_thoma15/spack/var/spack/repos/builtin/packages/linkphase3/package.py:39, in install:
     0         def install(self, spec, prefix):
     1             fortran = self.compiler.fc
  >> 2             fortran('LINKPHASE3.f90', '-o', 'LINKPHASE3')
     3             mkdirp(prefix.bin)
     4             install('LINKPHASE3', prefix.bin)
```

@tgamblin advised me to manually compile using `which()` in #5231, my reasoning for doing it the way I did. I did, however, add the `ifort` option in the case that the user is utilizing the intel compiler in this commit.

File: var/spack/repos/builtin/packages/linkphase3/package.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    url      = "https://github.com/tdruet/LINKPHASE3"
+
+    version('2017-06-14', commit='559913593fc818bb1adb29796a548cf5bf323827',
+            git='https://github.com/tdruet/LINKPHASE3.git')
+
+    def install(self, spec, prefix):
+        if self.compiler.name == 'intel':
+            fortran = which('ifort')
+        else:
+            fortran = which('gfortran')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/linkphase3/package.py]: just do 
```
fortran = Executable(self.compiler.fc)
```

**Comment by junghans** [comment]
@akthoma : Does my change work for you?

**Comment by akthoma** [comment]
@junghans: Yes, thank you!

**Pull request merged**